storage/reads/array_cursor.gen.go

```diff
@@ -286,9 +286,6 @@ type floatWindowLimitArrayCursor struct {
 }
 
 func newFloatWindowLimitArrayCursor(cur cursors.FloatArrayCursor, every int64) *floatWindowLimitArrayCursor {
-	if every == 0 {
```

Comment:
I would think that instead of moving this logic into the `Next()` method itself, you could just create a non-windowed cursor here?

Reply:
I do create a non-windowed cursor if `every == 0`, however I still check that `every != 0` inside `Next()` as that code will panic if the zero value is used.